### ISSUE TITLE
🐛 Multiple Pet Bug

### ISSUE THREAD

**Issue by UmeboshiXI** [DerpyProjectGroup/topaz#5441]
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Windows 10

## Branch affected by issue 

`base` (Unmodified LSB)

## Steps to reproduce
Set job to DRG/PUP, Use Call Wyvern. Use !changejob DRG 75. See that your pet commands have disappeared. Use Call Wyvern, see that you now have two wyverns following you.

Note: This works with any pet job with pup subbed. Only the most recently spawned pet has any logic/commands besides following the player.

Note: I have had players that were also able to trigger this intermittently by spamming abilities that spawn pets after zoning. Though the GM commands with job change is the easiest way to reproduce this. After zoning with multiple pets out, the extra pets seem to get cleared.

**Comment by MowFord** [comment]
I actually ran into this bug while testing the fix in [this](https://github.com/LandSandBoat/server/pull/5531)

The issue here appears to be due to `LoadAutomaton(` calling `LoadAutomatonStats(` and resetting PPet, but most pets don't check if their master has null PPet so the old wyvern is just hanging around.

![image](https://github.com/LandSandBoat/server/assets/131182600/a94320b9-0f86-4226-9ad9-57f097fea2b4)

The changeJob and changeSJob bindings have completely separate logic from the packet_handler that receives a job change packet. But I believe both have the oversight that they don't handle puppetmaster specifically very well and assume they don't have a pet.

Simplest solution might be to just cleanly remove a pet before changing jobs in both places.

**Comment by WinterSolstice8** [comment]
about ready to refactor automatons because of this spaghetti. I spent some time looking at the instance crash and it's related to something similar to this

tl;dr: 'ate PAutomaton

**Comment by FFXILevelDown** [comment]
wanted to ask if this was the cause of the automation skill reset on zoning before I open an issue. sql char_skills stays intact but when calling automation after a zone, his skill in equipment>automation is back to 0

**Comment by WinterSolstice8** [comment]
Closed by #6956
